### ISSUE TITLE
`prefer-https`: Usages of namespaces that are defined as HTTP URLs are reported as violations but should not

### ISSUE THREAD

**Issue by stof** [sindresorhus/eslint-plugin-unicorn#3072]
### Description

Many XML namespaces (including the SVG one for instance) are defined as `http` URL. Those cannot be replaced by `https` URLs but the rule reports violations on them

### Examples

```js
// ✅
const SVG_NAMESPACE = 'http://www.w3.org/2000/svg'
const svg = document.createElementNS(SVG_NAMESPACE, 'svg')
svg.setAttributeNS('http://www.w3.org/2000/xmlns/', 'xmlns', SVG_NAMESPACE)
```

**Comment by lkritsimas** [comment]
Just ran into this as well! It's connected to #3062

Here's a list of some potentially affected methods, there could be more.

### Document
* [`createAttributeNS()`](https://developer.mozilla.org/en-US/docs/Web/API/Document/createAttributeNS)
* [`createElementNS()`](https://developer.mozilla.org/en-US/docs/Web/API/Document/createElementNS)
* [`getElementsByTagNameNS()`](https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementsByTagNameNS)

### DOMImplementation
* [`createDocument()`](https://developer.mozilla.org/en-US/docs/Web/API/DOMImplementation/createDocument)

### Element
* [`getAttributeNodeNS()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getAttributeNodeNS)
* [`getAttributeNS()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getAttributeNS)
* [`getElementsByTagNameNS()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getElementsByTagNameNS)
* [`hasAttributeNS()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/hasAttributeNS)
* [`removeAttributeNS()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/removeAttributeNS)
* [`setAttributeNS()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttributeNS)

### NamedNodeMap
* [`getNamedItemNS()`](https://developer.mozilla.org/en-US/docs/Web/API/NamedNodeMap/getNamedItemNS)
* [`removeNamedItemNS()`](https://developer.mozilla.org/en-US/docs/Web/API/NamedNodeMap/removeNamedItemNS)

### Node
* [`lookupPrefix()`](https://developer.mozilla.org/en-US/docs/Web/API/Node/lookupPrefix)
* [`isDefaultNamespace()`](https://developer.mozilla.org/en-US/docs/Web/API/Node/isDefaultNamespace)
